Update usaco-991.mdx (#5669)
cpinitiative/usaco-guide: solutions/silver/usaco-991.mdx

id: usaco-991
source: USACO Silver 2020 January
title: Loan Repayment
author: Brad Ma, Raviteja Kompalli, Kevin Sheng, Arnav Gokhale
---

[Official Analysis (C++)](http://www.usaco.org/current/data/sol_loan_silver_jan20.html)

## Explanation

> Firstly, observe that we can binary search on the answer, or $X$ in this case. For each $X$, we check if it is valid and update our binary search accordingly.

Reviewer: ```suggestion
Firstly, notice that we can binary search on the answer, or $X$ in this case. For each $X$, we must check if it is a valid, and update the binary search accordingly.
```

also should briefly mention why (i.e. "well smaller x -> ur giving him more" etc)

Reviewer: idk why the suggestion renders weird, i was fixing a typo

Reviewer: oh guoggins diff LOL

> Notice that the constraint on $M$ is up to $10^{12}$, which leads us to believe that the solution will be in square root time. To more efficiently determine whether a given $X$ value is valid, we can simulate the process. However, we process all states where $Y$ does not change at once.

Reviewer: at least to me, $10^{12}$ doesn't automatically give away sqrt (altho maybe it should idk)

i think it's more natural that you'd figure out there's only sqrt distinct values of Y -> you realize that sqrt runs in time (?) could be wrong
